src/libraries/Common/tests/System/Net/Http/Http3LoopbackConnection.cs

```diff
@@ -49,10 +50,10 @@ public override void Dispose()
 
             if (!_closed)
             {
-                CloseAsync(H3_INTERNAL_ERROR).GetAwaiter().GetResult();
+            //    CloseAsync(H3_INTERNAL_ERROR).GetAwaiter().GetResult();
```

Comment:
Why was this commented out?

Reply:
This is a workaround for what we were talking about late last week.

Our current API usage is:

1. Write response to stream.
2. Shutdown stream and wait for shutdown to complete.
3. Close stream.
4. Close connection.

Observed behavior is that "wait for shutdown to complete" is not sufficient here as the client receives the CONNECTION_CLOSE frame and immediately aborts all of its streams even if they haven't finished reading. We need to put some though into correct usage; perhaps @nibanks has a recommendation